Nexd/Nexd/ui/onboarding/RegistrationViewController.swift

```diff
@@ -208,14 +239,50 @@ class RegistrationViewController: ViewController<RegistrationViewController.View
         confirmPasswordImageView.image = R.image.lock1()
         confirmPasswordImageView.contentMode = .scaleAspectFit
     }
+    private func drawCircle(on button: UIButton) {
+        let buttonWidth = button.frame.size.width
+        let buttonHeight = button.frame.size.height
+
+        let centerCoordinates = CGPoint(x: buttonWidth / 2, y: buttonHeight / 2)
+
+        let smallestAspect = min(button.frame.width, button.frame.height)
+        let circleRadiusWithinButton = smallestAspect / 2
+
+        // prevents setting the layer when the constraints have not been set properly yet
+        if buttonWidth != 0, buttonHeight != 0, circleRadiusWithinButton != 0 {
+            let circularPath = UIBezierPath(arcCenter: centerCoordinates,
+                                            radius: circleRadiusWithinButton,
+                                            startAngle: 0,
+                                            endAngle: 2 * CGFloat.pi,
+                                            clockwise: true)
+            caShapeLayer.path = circularPath.cgPath
+            caShapeLayer.strokeColor = R.color.nexdGreen()?.cgColor
+            caShapeLayer.lineWidth = 3.0
+            caShapeLayer.fillColor  = UIColor.clear.cgColor
+            button.layer.addSublayer(caShapeLayer)
+        }
+    }
 }
 
 extension RegistrationViewController {
+    @objc func confirmTermsOfUseButtonPressed() {
+        if caShapeLayer.fillColor == UIColor.clear.cgColor {
```

Comment:
we shouldn't use the `fillColor` value to decide on whether the privacy policy has been accepted or not, but instead tthe new variable you added (confirmTermsOfUseButtonPressed)